src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -168,60 +173,71 @@ export default class UBAFeeSpokeCalculator {
   }
 
   /**
-   * Calculates the fee for a simulated deposit operation
-   * @param amount The amount of tokens to deposit
-   * @param refundChainId The chain id of the refund spoke
-   * @param flowRange The range of the flow to simulate the deposit for. Defaults to undefined to simulate the deposit for the entire flow.
-   * @returns The fee for the simulated deposit operation
+   * A convenience method for resolving the event fee for the spoke and the given symbol and a given flow type
+   * @param amount The amount of tokens to simulate
+   * @param flowType The flow type to simulate
+   * @param flowRange The range of the flow to simulate the event for. Defaults to undefined to simulate the event for the entire flow.
+   * @returns The event fee for the spoke and the given symbol and a given flow type
    */
-  public getDepositFee(amount: BigNumber, refundChainId: number, flowRange?: UBAFlowRange): BigNumber {
-    let depositorFee = toBN(0);
-
-    // Resolve the alpha fee of this action
-    const alphaFee = this.config.getBaselineFee(this.chainId, refundChainId);
-
-    // Contribute the alpha fee to the LP fee
-    depositorFee = depositorFee.add(alphaFee);
+  protected getEventFee(amount: BigNumber, flowType: "inflow" | "outflow", flowRange?: UBAFlowRange): FlowFee {
+    // The rough psuedocode for this function is as follows:
+    // We'll need two inflow/outflow curves
+    // We need to determine which flow curve to use based on the flow type
+    // Compute first balancing fee <- f(x, x+amnt)
+    // Compute second balance fee (oportunity cost) <- g(x+amnt, x)
+    // Incentive Fee (LP Fee Component): first balancing fee - second balance fee
+    // Return (LP Fee + Balancing Fee)
+    // #############################################
+
+    // We'll need to now compute the concept of the running balance of the spoke
+    const { runningBalance } = this.calculateHistoricalRunningBalance(flowRange?.startIndex, flowRange?.endIndex);
+
+    // We first need to resolve the inflow/outflow curves for the deposit and refund spoke
+    const flowCurve = this.config.getBalancingFeeTuples(this.chainId);
+
+    // Next, we'll need to compute the first balancing fee from the running balance of the spoke
+    // to the running balance of the spoke + the amount
+    const mainFee = computePiecewiseLinearFunction(
+      flowCurve,
+      runningBalance,
+      amount.mul(flowType === "inflow" ? 1 : -1)
+    );
 
-    // Resolve the historical running balance of the spoke
-    const depositRunningBalance = this.calculateHistoricalRunningBalance(flowRange?.startIndex, flowRange?.endIndex);
+    // Next, we'll need to compute the reverse balancing fee. This is the opportunity cost of
+    // the LP fee holders to compute the opposite flow and to essentially reverse the flow
+    const reverseFee = computePiecewiseLinearFunction(
```

Comment:
I don't think we need this `reverseFee` anymore and therefore its no longer affecting the `lpFee`. See this [commit](https://github.com/UMAprotocol/UMIPs/pull/581/commits/7152e17bc9a192adc16af2961a91d293be05f5ef) which implements the simplified variables that Matt proposed offline

Reply:
Updated: https://github.com/across-protocol/sdk-v2/pull/217/commits/c31c655fc37d20307e203edb6da5f5206d937514